net/chrony/src/opnsense/service/templates/OPNsense/Chrony/chrony.keys

```diff
@@ -0,0 +1,21 @@
+{% if helpers.exists('OPNsense.chrony.general.enabled') and OPNsense.chrony.general.enabled == '1' %}
+
+{%   if helpers.exists('OPNsense.chrony.general.symmetric_auth') and OPNsense.chrony.general.symmetric_auth == '1' %}
+
+{%     if helpers.exists('OPNsense.chrony.general.symmetric_keys_file') and not helpers.empty('OPNsense.chrony.general.symmetric_keys_file') == '1' %}
+
+# This is the chrony keys file.  It enables authentication of NTP
+# packets with symmetric keys when its location is specified by the keyfile
+# directive in chrony.conf(5).  It should be readable only by root and the
+# user under which chronyd is running.
+#
+# Don't use example keys!  It's recommended to generate random keys using
+# the chronyc keygen command.
+
+{{ OPNsense.chrony.general.symmetric_keys_file }}
```

Comment:
Dump the multi-line text widget as-is.

Reply:
I'm afraid this goes against our coding guidelines... https://docs.opnsense.org/development/guidelines/basics.html#safeguard-user-input

What is the problem scope? Writing the file manually will overwrite it later or just convenience to store the value in the GUI?

Reply:
NTPD does the same thing.   What's the best way to make the chrony.keys file persistent?  Is there away to bypass the PHP UI?

Reply:
I want the chrony.keys file to be created each time the router boots up.   (Preferably WORM - write once.)

Reply:
@fichtner.  Is the ask to re-write chrony like services_ntpd.php?  (It seems like overkill.  Likewise, there is a risk of creating a tarpit.)

Reply:
Let’s slow down a bit. NTPd is not the scope a here and we don’t accept static PHP submission anymore anyway.

I’m going to reiterate:

Config blobs into files are discouraged, because they cannot be validated. You can split up the content in a model in these cases.

what is odder a that /etc/ntpd.keys is a file in use by ntpd, but not chrony? Is that correct? If it’s also used by chrony i’d say it’s problematic as well for other reasons.